org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/contentmodel/DTDDiagnosticsTest.java

```diff
@@ -642,11 +642,82 @@ public void testDTDNotFoundWithPUBLIC() throws Exception {
 				d(5, 4, 5, 21, XMLSyntaxErrorCode.ETagRequired)); // [4]
 	}
 
+	@Test
+	public void MSG_SPACE_REQUIRED_BEFORE_ELEMENT_TYPE_IN_ELEMENTDECL_DTD() throws Exception {

```

Comment:
I suggest to have just one test for diagnostic+code action.

Reply:
Please do that for other tests. the idea is to avoid having method test name which ends with CodeAction (ex: MSG_SPACE_REQUIRED_BEFORE_ELEMENT_TYPE_IN_ELEMENTDECLCodeAction). We should just have MSG_SPACE_REQUIRED_BEFORE_ELEMENT_TYPE_IN_ELEMENTDECL.